Try some parameter doc fixes

dd32 opened the pull request:
See #45.

Notes:
 - This is not mergable as-is, and is mostly untested.
 - Commit [Process lists within parameter blocks when deliberate br's are reta…](https://github.com/WordPress/wporg-developer/commit/249038c2dc524d3130c1bd6963327009b0537f83) relies upon this upstream change: https://github.com/WordPress/phpdoc-parser/compare/master...try/dev-45-newlines-in-docs

| Before | After |
| --- | --- |
| With Commit 1: [Parse lists within hash parameters.](https://github.com/WordPress/wporg-developer/commit/b7b60c323ade89245ce767cb5bf322b59b712663) |
| <img width="832" alt="Screen Shot 2022-06-01 at 4 47 26 pm" src="https://user-images.githubusercontent.com/767313/171344307-059a35ab-0052-4f62-93c7-5a462f3220eb.png"> | <img width="899" alt="Screen Shot 2022-06-01 at 4 11 42 pm" src="https://user-images.githubusercontent.com/767313/171344373-4f908f5f-a9b3-4311-b69a-45cebfd791f8.png"> |
| With Commit 2: [Process lists within parameter blocks when deliberate br's are reta…](https://github.com/WordPress/wporg-developer/commit/249038c2dc524d3130c1bd6963327009b0537f83) |
| See above | <img width="598" alt="Screen Shot 2022-06-01 at 4 39 40 pm" src="https://user-images.githubusercontent.com/767313/171344642-4d17a5a4-f3cb-4a58-ad34-2c784304c010.png"> |
| Better example of newline retention |
| <img width="772" alt="Screen Shot 2022-06-01 at 4 50 13 pm" src="https://user-images.githubusercontent.com/767313/171344728-2999754a-31cb-410e-927a-e65e417512d8.png"> | <img width="613" alt="Screen Shot 2022-06-01 at 4 50 38 pm" src="https://user-images.githubusercontent.com/767313/171344796-e5356b49-e9fb-4adc-949f-071daad00100.png"> |

dd32 committed:
Parse lists within hash parameters.

dd32 committed:
Process lists within parameter blocks when deliberate <br>'s are retained.

Relies upon https://github.com/WordPress/phpdoc-parser/compare/master...try/dev-45-newlines-in-docs

StevenDufresne commented:
Would this be related to this one? Currently on trunk:
[_wp_get_iframed_editor_assets](http://localhost:8888/reference/functions/_wp_get_iframed_editor_assets/)
![](https://d.pr/i/Td0y5s.png)

StevenDufresne commented:
>This is not mergable as-is, and is mostly untested.

What makes this not mergeable? Does this need some tests?

dd32 commented:
> What makes this not mergeable? 

https://github.com/WordPress/wporg-developer/commit/b7b60c323ade89245ce767cb5bf322b59b712663 / https://github.com/WordPress/wporg-developer/pull/49/commits/d1ff23b49c3b4c08545672751cfa2b2947ad82a6 can probably be merged

https://github.com/WordPress/wporg-developer/commit/249038c2dc524d3130c1bd6963327009b0537f83 / https://github.com/WordPress/wporg-developer/pull/49/commits/e1e4408ab7bbbdd499d14c26a620d40088b4577b ~however relies upon an upstream change within the `phpdoc-parser` plugin~ relies upon [48dfc75](https://github.com/WordPress/wporg-developer/pull/49/commits/48dfc752d93f5346e3a4b30784d569a999e8a845)

The path forward here I think is
 - Make a decision on if keeping newlines would be best (I think so)
 - Update the parser to replace newlines with `<br>` instead, potentially only `.\n` though as in  [48dfc75](https://github.com/WordPress/wporg-developer/pull/49/commits/48dfc752d93f5346e3a4b30784d569a999e8a845)~https://github.com/WordPress/phpdoc-parser/compare/master...try/dev-45-newlines-in-docs~
 - ~Ensure the public devhub works with the above change, then finalise these parts.~

dd32 committed:
Keep newlines in the PHPDoc description when they follow a full-stop.

This allows for intentionally-written newlines to be available to developer.wordpress.org, otherwise they're removed entirely.

This is needed for 249038c

dd32 committed:
Insert a newline between line items when parsing.

dd32 committed:
Use a much simpler list items conversion process.
